client/src/utils/localization.js

```diff
@@ -42,6 +24,56 @@ define(["i18n!nls/locale"], function (localeStrings) {
     };
     localize.cacheNonLocalized = false;
 
+    localize._setUserLocale = function _setUserLocale(user, config) {
+        // Choose best locale
+        const global_locale =
+            config.default_locale && config.default_locale != "auto" ? config.default_locale.toLowerCase() : false;
+
+        let extra_user_preferences = {};
+        if (user && user.attributes.preferences && "extra_user_preferences" in user.attributes.preferences) {
+            extra_user_preferences = JSON.parse(user.attributes.preferences.extra_user_preferences);
+        }
+
+        let user_locale =
+            "localization|locale" in extra_user_preferences
+                ? extra_user_preferences["localization|locale"].toLowerCase()
+                : false;
+
+        if (user_locale == "auto") {
+            user_locale = false;
+        }
+
+        const nav_locale =
+            typeof navigator === "undefined"
+                ? "__root"
+                : (navigator.language || navigator.userLanguage || "__root").toLowerCase();
+
+        const locale = user_locale || nav_locale || global_locale;
+
+        sessionStorage.setItem("currentLocale", locale);
+    };
+    // =============================================================================
+    /** Simple string replacement localization. Language data from src/nls */
+
+    localize._getUserLocale = function _getUserLocale(user, config) {
+        let locale = null;
+        if (Object.prototype.hasOwnProperty.call(localeStrings, "__root")) {
+            //console.debug( 'amdi18n+webpack localization for ' + locale + ' loaded' );
```

Comment:
@hujambo-dunia do you plan to clean this up before merging?

Reply:
@bgruening yes - for one thing, there were a bunch of comments left by prior developer which I would like to remove.